models/wishlist.model.js

```diff
@@ -0,0 +1,20 @@
+const mongoose = require("mongoose");
+const { User } = require("../models/user.model");
+const { Product } = require("../models/product.model");
+const WishlistSchema = new mongoose.Schema(
```

Comment:
Here too, please check the review for the cart model, **If you change this schema you also have to make changes to route and frontend. Your schema works too, this is just an improvement**
schema can be improved as
```
const WishlistSchema = new Schema({
    user: {
        type: Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    wishlist: [
        {type: Schema.Types.ObjectId, ref: "Product", required: true}
    ]
})
```